scripts/enforce_match_arg_in_assert_produces_warning.py

```diff
@@ -0,0 +1,83 @@
+"""
+
+Enforce that all usages of tm.assert_produces_warning use
+the "match" argument. This will help ensure that users always see
+the correct warning message.
+
+tm.assert_produces_warning(None), tm.assert_produces_warning()
+and tm.assert_produces_warning(False) are excluded as no warning is
+produced.
+
+"""
+
+
+import argparse
+import ast
+from collections.abc import Sequence
```

Comment:
```suggestion
from typing import TYPE_CHECKING
...
if TYPE_CHECKING:
    from collections.abc import Sequence
```

Reply:
Had to remove this one, caused build failure

Reply:
You should remove the line with 3 dots 😅

Reply:
Yes that I did. But it's still unable to import `Sequence`